lib/core.sh

```diff
@@ -396,13 +401,10 @@ function build_image_env(){
     sudo bash -c "echo 'export PATH=/opt/yaourt/bin:\$PATH' > ${maindir}/root/etc/profile.d/${CMD}.sh"
     sudo chmod +x ${maindir}/root/etc/profile.d/${CMD}.sh
 
-    info "Install ${NAME} script..."
-    sudo pacman --noconfirm --root ${maindir}/root -S git
-    _install_from_aur ${maindir} "${CMD}-git" "${CMD}.install"
-    sudo pacman --noconfirm --root ${maindir}/root -Rsn git
-
     info "Setting up the pacman keyring (this might take a while!)..."
-    sudo arch-chroot ${maindir}/root bash -c "pacman-key --init; pacman-key --populate archlinux"
+    sudo pacman --noconfirm --root ${maindir}/root -S psmisc
+    sudo ${maindir}/root/opt/junest/bin/jchroot ${maindir}/root bash -c "pacman-key --init; pacman-key --populate archlinux; killall gpg-agent"
```

Comment:
Good point indeed. The major problem here is that the gpg-agent's parent is the process with PID 1 and it looks hard to identify which is the right process to kill:

    ps -ef | grep gpg-agent
    root     19960     1  0 11:10 ?        00:00:00 gpg-agent --homedir /etc/pacman.d/gnupg --use-standard-socket --daemon

From the `pacman-key` point of view it looks that killing the gpg-agent is harmless since it will spawn the agent in case it does not exist. Do you know a way to tackle this differently?